indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -737,14 +1038,35 @@ private boolean generateAndPublishSegments(
                 )
             )
         );
-        return true;
+
+        return TaskStatus.success(
+            getId(),
+            getTaskCompletionMetrics(),
+            null,
+            getTaskCompletionContext()
+        );
       }
     }
     catch (TimeoutException | ExecutionException e) {
       throw Throwables.propagate(e);
     }
   }
 
+  @Nullable
+  public static List<String> getMessagesFromSavedParseExceptions(CircularBuffer<Throwable> savedParseExceptions)
```

Comment:
IndexTask class is already very long enough. Suggest to extract as a method of a separate util class.